src/js/angular/ttyg/directives/chat-panel.directive.js

```diff
@@ -64,10 +64,21 @@ function ChatPanelComponent(toastr, $translate, TTYGContextService, TTYGService)
                 if (!$scope.chatItem.chatId) {
                     createNewChat();
                 } else {
-                    askQuestion();
+                    askQuestion($scope.chatItem);
                 }
             };
 
+            $scope.regenerate = (chatItem) => {
+                const question = getEmptyChatItem();
+                question.question.message = chatItem.question.message;
+                const regenerateChatItem = new ChatItemModel(chatItem.chatId, chatItem.agentId, question.question);
```

Comment:
What's the point to create a new ChatItemModel just to get the question from it?!

Reply:
Oops, this is an old implementation from before I asked which agentId to use: the chat item's agent or the selected agent. I forgot to update this part of code.